pkg/imageregistry/imageregistry_test.go

```diff
@@ -0,0 +1,379 @@
+		expectedError     error
+		managementState   operatorV1.ManagementState
+	}{
+		{
+			testImageRegistry: buildValidImageRegistryBuilder(buildImageRegistryClientWithDummyObject()),
+			expectedError:     nil,
+			managementState:   operatorV1.Managed,
+		},
+		{
+			testImageRegistry: buildInValidImageRegistryBuilder(buildImageRegistryClientWithDummyObject()),
+			expectedError:     fmt.Errorf("the imageRegistry 'name' cannot be empty"),
+			managementState:   operatorV1.Managed,
+		},
+	}
+
+	for _, testCase := range testCases {
+		assert.Equal(t, defaultManagementState, testCase.testImageRegistry.Definition.Spec.ManagementState)
+		assert.Nil(t, nil, testCase.testImageRegistry.Object)
+		testCase.testImageRegistry.WithManagementState(testCase.managementState)
+		_, err := testCase.testImageRegistry.Update()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, testCase.managementState, testCase.testImageRegistry.Definition.Spec.ManagementState)
+		}
+	}
+}
+
+func TestImageRegistryWithManagementState(t *testing.T) {
+	testCases := []struct {
+		testManagementState operatorV1.ManagementState
+		expectedError       bool
+		expectedErrorText   string
+	}{
+		{
+			testManagementState: operatorV1.Removed,
+			expectedError:       false,
+			expectedErrorText:   "",
+		},
+		{
+			testManagementState: operatorV1.Managed,
+			expectedError:       false,
+			expectedErrorText:   "",
+		},
+		{
+			testManagementState: operatorV1.Unmanaged,
+			expectedError:       false,
+			expectedErrorText:   "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidImageRegistryBuilder(buildImageRegistryClientWithDummyObject())
+
+		result := testBuilder.WithManagementState(testCase.testManagementState)
+
+		if testCase.expectedError {
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, result.errorMsg)
+			}
+		} else {
+			assert.NotNil(t, result)
+			assert.Equal(t, testCase.testManagementState, result.Definition.Spec.ManagementState)
+		}
+	}
+}
+
+func TestImageRegistryGetManagementState(t *testing.T) {
+	testCases := []struct {
+		testImageRegistry *Builder
+		expectedError     error
+	}{
+		{
+			testImageRegistry: buildValidImageRegistryBuilder(buildImageRegistryClientWithDummyObject()),
+			expectedError:     nil,
+		},
+		{
+			testImageRegistry: buildValidImageRegistryBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:     fmt.Errorf("imageRegistry object doesn't exist"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		currentManagementState, err := testCase.testImageRegistry.GetManagementState()
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, currentManagementState, testCase.testImageRegistry.Object.Spec.ManagementState)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestImageRegistryWithStorageConfig(t *testing.T) {
```

Comment:
The function under test is WithStorage and Test function is TestImageRegistryWithStorageConfig. It would be hard to find what cause the failure in unit test. 
Can you please either re-name WithStorage function to WithStorageConfig OR
Rename test function  from TestImageRegistryWithStorageConfig to TestImageRegistryWithStorage.